client/webserver/site/src/html/markets.tmpl

```diff
@@ -195,6 +195,21 @@
           <div class="order-panel pb-3 position-relative">
             <div class="d-flex flex-column align-items-stretch">
 
+              {{- /* TOKEN APPROVAL */ -}}
+              <div class="fs15 pt-1 pb-3 text-center brdrbottom d-hide" id="tokenApproval">
+                <span class="p-3 flex-center fs16 grey d-hide" id="approvalRequiredBuy">[[[approval_required_buy]]]</span>
+                <span class="p-3 flex-center fs16 grey d-hide" id="approvalRequiredSell">[[[approval_required_sell]]]</span>
+                <span class="p-3 flex-center fs16 grey d-hide" id="approvalRequiredBoth">[[[approval_required_both]]]</span>
+                <button id="approveBaseBttn" type="button" class="pointer hoverbg text-center p-2 text-nowrap">[[[Approve]]] <span id="baseTokenAsset"></span></button>
+                <div id="approvalPendingBase" class="d-hide flex-center position-relative py-2">
+                  <span class="px-1" id="approvalPendingBaseSymbol"></span> Approval Pending <div class="px-2 ico-spinner spinner fs15"></div>
+                </div>
+                <button id="approveQuoteBttn" type="button" class="pointer hoverbg text-center p-2 text-nowrap">[[[Approve]]] <span id="quoteTokenAsset"></span></button>
+                <div id="approvalPendingQuote" class="d-hide flex-center position-relative py-2">
+                  <span class="px-1" id="approvalPendingQuoteSymbol"></span> Approval Pending <div class="px-2 ico-spinner spinner fs15"></div>
```

Comment:
The "Approval Pending" will have to go in the locales, either backend html renderer or the javascript.

I also noticed that after unapproving, it said "Approval Pending", which is not wrong since it's approving 0, but could be confusing.  Maybe "Approval Change Pending"?